### PR TITLE
Add filter command

### PR DESCRIPTION
**Pull request by yiwen101**
Screenshot 2023-11-02 at 21 14 13
Screenshot 2023-11-02 at 21 14 16
Screenshot 2023-11-02 at 21 14 57
Screenshot 2023-11-02 at 21 15 00
Screenshot 2023-11-02 at 21 15 12
Screenshot 2023-11-02 at 21 15 25
Screenshot 2023-11-02 at 21 26 04
Screenshot 2023-11-02 at 21 20 57
Screenshot 2023-11-02 at 21 21 01
Screenshot 2023-11-02 at 21 22 01
Screenshot 2023-11-02 at 21 22 05

### CODE REVIEW

**Comment by yiwen101** [comment]
After debug for duplicate flag:
Screenshot 2023-11-02 at 22 39 54

File: src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public EditLessonCommand parse(String args) throws ParseException {
 
     public static String getUsageInfo() {
         return "\nUsage: edit <Index> (at least one of -[name|subject|day|start|end|remark] [value]). "
-                + "\nFor example, edit 1 -name lesson2 -subject Math -day Monday -start 14:30 -end 16:30"
+                + "\nFor example, edit 1 -name lesson2 -subject English -day 23/12 -start 14:30 -end 16:30"
```
Comment (richiehx) [inline src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java:42]: The day here is a little different from the one in AddLessonCommand parser. Which one is the more suitable one?

**Review by richiehx — APPROVED** [review]
LGTM! Some minor questions.

File: src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java (hunk 1 above)
Comment (yiwen101) [reply src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java:42]: I also change the day in the usage for AddLessonCommand parser.
"-day Monday" is wrong and will throw error

Comment (yiwen101) [reply src/main/java/seedu/address/logic/parser/EditLessonCommandParser.java:42]: Currently only accept  yyyy/mm/dd or mm/dd or dd, will improve error message for Day in next PR

**Comment by pzl111** [comment]
> After debug for duplicate flag: Screenshot 2023-11-02 at 22 39 54

would it be clearer to say that the user can input multiple but unique flags?
(ie. -subject -tag is allowed, but -subject -subject is not)

**Comment by yiwen101** [comment]
> > After debug for duplicate flag: Screenshot 2023-11-02 at 22 39 54
> 
> would it be clearer to say that the user can input multiple but unique flags? (ie. -subject -tag is allowed, but -subject -subject is not)

Good point, will add the word unique

**Comment by yiwen101** [comment]
Message improved in the next PR, which is on a branch focused on improving message

**Pull request merged**